include/xsimd/types/xsimd_avx512_int16.hpp

```diff
@@ -333,6 +333,12 @@ namespace xsimd
                 XSIMD_RETURN_MERGED_AVX(res_lo, res_hi);
             #endif
             }
+
+            template<bool... Values>
+            static batch_type select(const batch_bool_constant<value_type, Values...>& cond, const batch_type& a, const batch_type& b)
+            {
+              return select(cond(), a, b);
+            }
```

Comment:
Since the select method has the same implementation for all the batches of int, I wonder if it could be moved to `avx512_int_kernel_base` to avoid code duplication.

I understand we cannot have a generic implementation in `simd_base` since some implementation uses specific intrinsics (not the case for AVS512 though).